app/src/main/kotlin/com/wire/android/ui/home/gallery/MediaGalleryViewModel.kt

```diff
@@ -54,9 +71,53 @@ class MediaGalleryViewModel @Inject constructor(
         mediaGalleryViewState = mediaGalleryViewState.copy(screenTitle = conversationName)
     }
 
+    fun saveImageToExternalStorage() {
+        viewModelScope.launch {
+            withContext(dispatchers.io()) {
+                val imageData = getImageData(imageAssetId.conversationId, imageAssetId.messageId)
+                if (imageData is Success) {
```

Comment:
What if not success? We just implement happy path for now?

Reply:
Good catch. I will add an error message in case the download fails 👍